gitnamehere/point-of-sale-group-project
# Cart display

**Ticket:** [Cart Display](https://cs375-pos-group-project.atlassian.net/jira/core/projects/ATS/board?selectedIssue=ATS-59)

**PR Desc:**
- Added API endpoints for `cart_item` in `api.js` 
- - **GET API endpoint:** Returns all cart items
- - **PUT API endpoint:** Update quantity of a specific cart item
- - **DELETE API endpoint:** Removes specific items from the cart
- Added a toast/notification that appears when you add items to the cart
- Added a modal that appears when you click on the cart button in the navbar
- - Displays all cart items with their price and quantity
- - Quantity can be adjusted via input or increment and decrement buttons
- - Displays total with the tax calculated at the bottom of the cart
- Added a close and place order button at the bottom of the cart
- - Added a modal to confirm the order when place order is clicked

**Testing:**
- Run `npm i`
- Run `npm run setup-demo:dev`
- Run `npm start`
- Visit http://localhost:3000/store/menu/#
- Verify that a toast/notification appears at the bottom right when you add an item to the cart
- Add multiple items and remember the order you added them
- Verify that the items you add to the cart appear when you click the cart button in the navbar
- Verify that they appear in the order you added them in
- Remove one item and verify it is no longer in the cart
- Verify the total at the bottom of the cart is accurate
- Change the quantity of an item and ensure the total is also changing with accuracy
- Place an order and visit http://localhost:3000/pos/orders/ to see if your order appears
- Attempt to place an order with no items in the cart and verify it tells you that you can't
